lms/djangoapps/courseware/views/index.py

```diff
@@ -446,6 +452,32 @@ def _create_courseware_context(self, request):
         )
         staff_access = self.is_staff
 
+        reset_deadlines_url = reverse(
+            'openedx.course_experience.reset_course_deadlines', kwargs={'course_id': six.text_type(self.course.id)}
+        )
+
+        allow_anonymous = allow_public_access(self.course, [COURSE_VISIBILITY_PUBLIC])
+
+        display_reset_dates_banner = False
+        if not allow_anonymous:  # pylint: disable=too-many-nested-blocks
+            course_overview = CourseOverview.objects.get(id=str(self.course_key))
+            end_date = getattr(course_overview, 'end_date')
```

Comment:
I think you can just use `self.course.end_date` here. If you already have the course (which is either a `CourseOverview` or a `CourseModule`), then you can just read the attribute off it directly.

Reply:
i remember when originally tryin that, that self,.course is a CourseMixinWithDescriptor object....and doesnt have an end_date prop.  just double checked and its indeed the case.